### PR TITLE
Added the posibility to change fuel consumption for all drives.

### PR DESCRIPTION
**Pull request by hdhdhd95**
It also allows a drive to have a lower consumption hyper/scram drive mode and a jump drive mode.

### CODE REVIEW

**Comment by KevinKorzekwa** [comment]
Not trying to suggest a problem, I am simply curious.  Why change the return attributes in the Ship::JumpFuel function from hundreds to tens?  This is line 1740 to 1742 of the changed file.  

What is the change made by doing that?

**Comment by hdhdhd95** [comment]
I was trying to figure out how the values work and tried making all travel 1/10 as expensive in terms of fuel but it didn't work and I forgot to change it back.

**Comment by hdhdhd95** [comment]
I now set the values mentioned by @KevinKorzekwa back to their previous state and incorporated a tentative of issue 1455

**Comment by Wrzlprnft** [comment]
i'm not sure about the korath jump drive, if it should be a thing.

**Comment by KiLEdEnNis** [comment]
indeed, the Korath "plundered" them from the Quarg, who got them from the Drak, not like the Quarg would change them
and it'll conflict with the Unfettered Hai Jump Drive mission thing

File: data/drives.txt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	category "Systems"
+	cost 80000
+	thumbnail "outfit/hyperdrive"
+	"mass" 20
+	"outfit space" -20
+	"hyperdrive fuel" 120
+	"jump speed" 10
+	"hyperdrive" 1
+	description "This mysterious box sits quietly at the heart of your ship, thinking its own inscrutable thoughts. When it comes time to travel between the stars it sets off a nuclear fusion reaction powerful enough to level an entire city, and then somehow tucks the resulting explosion in between the folds of regular space-time like a wasp stinging an oak leaf to lay her eggs inside. Your ship rides the resulting ripple in the fabric of space."
+	description "	They say that fewer than a dozen human beings alive today understand the inner workings of the hyperdrive. This one is a Korath version faster but consumes a bit more fuel."
```
Comment (Bladewood) [inline data/drives.txt]: This one is a Korath version: faster but less fuel-efficient.

**Comment by Bladewood** [comment]
These drives are REALLY unbalanced.

**Comment by KiLEdEnNis** [comment]
@Bladewood then change them (if you can), some ideas are bad at first and become good through changing

File: data/drives.txt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	category "Systems"
+	cost 80000
+	thumbnail "outfit/hyperdrive"
+	"mass" 15
+	"outfit space" -10
+	"hyperdrive fuel" 120
+	"jump speed" .4
+	"hyperdrive" 1
+	description "This mysterious box sits quietly at the heart of your ship, thinking its own inscrutable thoughts. When it comes time to travel between the stars it sets off a nuclear fusion reaction powerful enough to level an entire city, and then somehow tucks the resulting explosion in between the folds of regular space-time like a wasp stinging an oak leaf to lay her eggs inside. Your ship rides the resulting ripple in the fabric of space."
+	description "	They say that fewer than a dozen human beings alive today understand the inner workings of the hyperdrive. This one is a Korath version faster but consumes a bit more fuel."
```
Comment (toilethinges) [inline data/drives.txt]: This one is a Korath version - faster, and takes less space, but consumes a bit more fuel."

**Comment by toilethinges** [comment]
I'd say maybe just implement the ability itself, then let MZ decide what stats (if any different) he wants to give the drives.

**Comment by hdhdhd95** [comment]
The ability is implemented and the reason I made the jump drive have such high speed is because the effect looks cooler but you can pull request a better balanced version and maybe add some of the designs I have been seeing they look pretty cool.

**Comment by KiLEdEnNis** [comment]
@hdhdhd95 having things look cool is ok, but don't x10 a stat :P

**Comment by warp-core** [comment]
@hdhdhd95 I was just looking through this pr and I noticed that commits I made on a fork of the ES master repository for a pr of my own are in this one, which is confusing to say the least. Can anyone suggest any way this could have happened because I don't understand it and they should probably be removed.

**Comment by hdhdhd95** [comment]
I was testing to make sure they wouldn't break anything.

**Comment by hdhdhd95** [comment]
@KiLEdEnNis now they are a bit more balanced.

**Comment by Wrzlprnft** [comment]
You should look into githubs branching. You can use 2 different branches for your projects

**Comment by hdhdhd95** [comment]
@Wrzlprnft I actually know how create branches.

**Comment by Wrzlprnft** [comment]
My bad, sorry. It is just that very feature/change heavy pull requests are less likely to get merged.

Keep up the good work

**Comment by Amazinite** [comment]
> The Korath first adquired Jump Drives by raiding the Quarg but with the years they figured out how to make their own.

Pretty sure that breaks lore. I'd honestly say get rid of the two different Jump Drives and just leave it at the current Jump Drive. From what we can tell, only the Quarg can create Jump Drives (if we exclude the Pug and their space magic), knowledge given to them by the Drak (if I recall correctly). And given the tier difference between the Korath and the Quarg, I don't think they would be able to reverse engineer them.

File: data/drives.txt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	thumbnail "outfit/jump drive"
+	"mass" 30
+	"outfit space" -30
+	"jump speed" 2.5
+	"jump drive" 1
+	"jump drive fuel" 150
+	"scram drive fuel" 75
+	"scram drive" 1
+	"hyperdrive" 1
+	description "The workings of the Jump Drive are a total mystery. The best that human scientists have been able to find out is that this device is a containment field for matter in a state that cannot normally exist in this universe; if the drive is opened up, the field collapses and its contents disappear. Therefore, even though a few jump drives have been captured over the years, it has so far proved impossible to reverse engineer them. This Jump Drive has been captured from a Korath ship. Like everything Quarg it is efficient."
```
Comment (Amazinite) [inline data/drives.txt]: "This Jump Drive has been captured from a Korath ship." Should be Quarg.

**Comment by warp-core** [comment]
@MarcelineVQ I noticed that in your recent PR you were seemingly able to remove commits (I believe they were the same change as mine, 'spare' to 'sparse' in the Varu K'est description). Could you please explain the process behind this because I can't figure this out and, evidently, neither can @hdhdhd95 and the commits from my fork and PR (still don't know how they got here, any insight you may have would be helpful) should probably be removed since MZ already explained why they aren't necessary.
I'm not very good with Github but, from what I've read, adding a commit that reverses the changes seems to be the best way, but again, I'm not very good with Github and you seem to have far more experience.

**Comment by MarcelineVQ** [comment]
I do that through rebasing. For example `git rebase -i HEAD~5` will let you modify the last 5 commits in various ways which it will tell you about when you do it. Because this causes the branches to diverge you'll need to force `git push -f` your push to github if you rebase.
It's important to only do this for independent things like pull requests or personal projects, rebasing branches that other people are working from can cause them issues. In general it's safest to make your changes as a new commit instead of rebasing but rebases are tidy.

**Comment by KevinKorzekwa** [comment]
@Wrzlprnft touched on this a few days ago, but I'd like to give more detail (at least as I see it).  At first glance, it seems there are too many things going on in this PR.

1) You add the alien drive outfits.  
__a) The alien drives are mission rewards.
__b) They are added to alien ship variants
__c) They are added to the alien outfitters.
2) You add a transport job.
3) You make a map correction (spare to sparse) that @endless-sky has said he does not want in #1470 and #1421.
4) You make it possible to change fuel consumption.
5) You change a for loop to a while loop (I admit I don't understand why yet.)

So, to sum up: 1, 2, and 3 are all changes to data files, while 4 and 5 are changes to the code.   A lot of stuff here, great stuff, but it is a lot to go through.  I'm going to ignore 3 because it has been discussed elsewhere, and I'm sure you can find those discussions to decide whether or not its presence here harms your PR or not.  I will also ignore 5, since I'm not clear on why you made that change.

1, 1b, 1c, and 4 seem like they could stand alone in a branch that deals with just adding the alien drives (and making it possible for the fuel consumption to be variable).   1a and 2 I would have put in a separate branch, since it crosses the line from adding them to giving them directly to the player.  It's a tough thing though, I could just as easily see putting 1 and 4 in one branch, and 1a, 1b, 1c, and 2 all in the other branch.  I guess that just depends on how it matches your brain.

Anyway, awesome work here.  I'm not trying to tell you take half of the stuff here and put it in another PR, just trying to give you input on how I interpret all the stuff in here.  Definitely keep up the great work!

**Comment by hdhdhd95** [comment]
I removed everything but the fuel consumption change from the branch.

**Comment by omagasohe** [comment]
you made a little typo...

`return (hasJumpDriveFuel ? hasJumpDriveFuel: 150 );`
should be 
`return (hasJumpDriveFuel ? hasJumpDriveFuel: 200 );`

**Comment by endless-sky** [comment]
Addressed in a different way in ebb46d8.

**Pull request closed without merging**